stdlib/importlib/abc.pyi

```diff
@@ -93,9 +85,9 @@ class FileLoader(ResourceLoader, ExecutionLoader, metaclass=ABCMeta):
 
 class ResourceReader(metaclass=ABCMeta):
     @abstractmethod
-    def open_resource(self, resource: StrOrBytesPath) -> IO[bytes]: ...
+    def open_resource(self, resource: str) -> IO[bytes]: ...
```

Comment:
[The documentation states](https://docs.python.org/3.11/library/importlib.resources.abc.html#importlib.resources.abc.ResourceReader):

> For any of methods of this class, a resource argument is expected to be a [path-like object](https://docs.python.org/3.11/glossary.html#term-path-like-object) which represents conceptually just a file name.

That would indicate that the methods on this class can indeed accept pathlike objects, and that it's CPython/`importlib_resources` that have incorrect annotations for this class.

Reply:
First of all, if this method should accept bytes, then `TraversableResources` violates the Liskov substitution principle.

The documentation goes on to say:

> This means that no subdirectory paths should be included in the resource argument.

Which is very slightly odd. This method accepts path-like objects but only if they do not represent paths at all. I think it is safer to conceal this misfeature from the typing user.

Reply:
> First of all, if this method should accept bytes, then `TraversableResources` violates the Liskov substitution principle.

Well, it would be _far_ from the first class in the stdlib to do so :)

Reply:
It does seem like the annotations and docstring for `importlib_resources` are completely different, and always have been: https://github.com/python/importlib_resources/blob/main/importlib_resources/abc.py.